src/transactions/TransactionFrameBase.h

```diff
@@ -56,8 +54,23 @@ class TransactionFrameBase
     insertKeysForFeeProcessing(UnorderedSet<LedgerKey>& keys) const = 0;
     virtual void insertKeysForTxApply(UnorderedSet<LedgerKey>& keys) const = 0;
 
-    virtual void processFeeSeqNum(AbstractLedgerTxn& ltx, int64_t baseFee) = 0;
+    virtual void processFeeSeqNum(AbstractLedgerTxn& ltx, int64_t baseFee,
+                                  TransactionResult& txResult) = 0;
 
     virtual StellarMessage toStellarMessage() const = 0;
+
+#ifdef BUILD_TESTS
```

Comment:
I haven't examined each usage of this, but it seems like you create test tx frames using the special method (`makeTestTransactionFromWire`). Would it be possible then to not have these methods in the public interface and only declare/implement them in the respective test-only classes?

Reply:
That would be difficult, as many of the existing tests are calling those public methods via the base class pointer. To remove those test-only functions from the base class to the "Test" classes would require updating all (hundreds of) callers in the test code to pass in the additional parameters.

Reply:
This is not entirely accurate.

See my other comment on `TransactionTestFrame`: I think that we should just use `TransactionTestFrame` in all test code instead of returning `TransactionFrameBase` or `TransactionFrame`. The change should be fairly transparent as we usually do things like
`auto b1 = root.create("b1", paymentAmount);` (that would return a `TransactionTestFrame` instead).

That way we can also move the other "mutators" from TransactionFrame, like `addSignature` that should only be used in test/simulation code.